R/derive_param_lasta.R

```diff
@@ -0,0 +1,358 @@
+#'   "03",     lubridate::ymd("2021-01-09"), lubridate::ymd("2021-02-24"),
+#'   "04",     lubridate::ymd("2021-04-21"), lubridate::ymd("2021-09-15")
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' adrs <- tibble::tribble(
+#'   ~USUBJID, ~PARAMCD, ~AVAL, ~AVALC, ~ASEQ, ~ADT,
+#'   "01", "RSP", NA, "Y", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "RSP", NA, "N", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "RSP", NA, "N", 1, NA,
+#'   "04", "RSP", NA, "N", 1, NA,
+#'   "01", "PD",  NA, "N", 1, NA,
+#'   "02", "PD",  NA, "Y", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "PD",  NA, "N", 1, NA,
+#'   "04", "PD",  NA, "N", 1, NA,
+#'   "01", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "01", "OVR", 2, "PR", 1, lubridate::ymd("2021-04-08"),
+#'   "02", "OVR", 3, "SD", 1, lubridate::ymd("2021-03-07"),
+#'   "02", "OVR", NA, NA, 1, lubridate::ymd("2021-04-07"),
+#'   "02", "OVR", 6, "PD", 1, lubridate::ymd("2021-05-07"),
+#'   "03", "OVR", 3, "SD", 1, lubridate::ymd("2021-01-30"),
+#'   "03", "OVR", 3, "SD", 2, lubridate::ymd("2021-01-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-05-21"),
+#'   "04", "OVR", 5, "NON-PD", 1, lubridate::ymd("2021-06-30"),
+#'   "04", "OVR", NA, "NE", 1, lubridate::ymd("2021-07-24"),
+#'   "04", "OVR", NA, "ND", 1, lubridate::ymd("2021-09-30"),
+#' ) %>%
+#'   dplyr::mutate(STUDYID = "a_study_id")
+#'
+#' pd <- date_source(
+#'   dataset_name = adrs,
+#'   date         = ADT,
+#'   filter       = PARAMCD == 'PD' & AVALC == 'Y'
+#' )
+#'
+#' derive_param_lasta(
+#'   dataset,
+#'   filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
+#'   source_pd = list(pd),
+#'   source_datasets = list(adrs = adrs),
+#'   set_values_to = vars(
+#'     PARAMCD = "LSTAC",
+#'     PARAM = "Last Disease Assessment Censored at First PD by Investigator",
+#'     PARCAT1 = "Tumor Response",
+#'     PARCAT2 = "Investigator",
+#'     PARCAT3 = "Recist 1.1",
+#'     ANL01FL = "Y")
+#' )
+#'}
+#'
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @export
+#' @name   derive_param_lasta
+#' @title  derive_param_lasta
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @author Stephen Gormley
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @keywords ADRS
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @importFrom  magrittr `%>%`
+#+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++|
+#' @return The dataframe passed in the `dataset` argument with additional
+#'         columns and/or rows as set in the `set_values_to` argument.
+#%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%|
+
+derive_param_lasta <- function(dataset,
+                               dataset_adsl,
+                               filter_source,
+                               order           = admiral::vars(STUDYID, USUBJID, ADT),
```

Comment:
lets see what Thomas replies around the "admiral::" convention

Reply:
disagree, but will do it, no problem.

Reply:
thanks!

Reply:
@bundfussr can you double check these please? i saw some `lubridate::` cases in some other devel codes that got merged like some of the unit tests, so i'm not actually sure which of these Stephen left in are acceptable or not

Reply:
- In the function code itself we do not specify the package for function calls. If a function of another package is called it is imported via `admiralxxx-package.R`.
- In the examples we load all required packages as the first step of the examples (and do not specify the package for function calls in the example code).
- In the tests we are less strict. So we have both `library()` calls at the beginning of the test and function calls with package name. Maybe something to discuss in the Wednesday meeting.

I also wonder if we should extend "R Package Dependencies" section in the programming strategy.

Reply:
yes think worthwhile Stefan as i was getting confused here